src/TraceEvent/EventPipe/EventPipeMetadata.cs

```diff
@@ -884,11 +886,28 @@ private void ExtractImpliedOpcode()
                 if (EventName.EndsWith("Start", StringComparison.OrdinalIgnoreCase))
                 {
                     Opcode = (int)TraceEventOpcode.Start;
-                    EventName = EventName.Substring(0, EventName.Length - 5);
                 }
                 else if (EventName.EndsWith("Stop", StringComparison.OrdinalIgnoreCase))
                 {
                     Opcode = (int)TraceEventOpcode.Stop;
+                }
+            }
+        }
+
+        /// <summary>
+        /// If the event has a Stop/Start opcode and also ends with the word "Start" or "Stop", then
+        /// remove the "Start" or "Stop" from the event name.
+        /// </summary>
+        private void StripStartStopInEventName()
+        {
+            if (EventName != null)
+            {
+                if (Opcode == (int)TraceEventOpcode.Start && EventName.EndsWith("Start", StringComparison.OrdinalIgnoreCase))
```

Comment:
Do we know that ActivityJson/Start and Activity1Start and Activity2Start also use the TraceEventOpcode.Start? (and respective stops use TraceEventOpcode.Stop)?

Reply:
They will all end with Start and Stop so should work correctly.

Reply:
EventSource internally attempts to set the Start/Stop opcode for any event that has "Start" or "Stop" as a name suffix and didn't explicitly set the opcode to something else. I'd expect the events you mentioned to all have the Start/Stop opcode set when emitted from a recent runtime version. I think old runtimes like .NET 3 and 5 didn't include the opcode when the data was serialized (the nettrace V4 format) but the parser should still handle it correctly regardless. The unit tests cover both when the opcode is set explicitly and when it is left at the default value.